src/AutoMapper/QueryableExtensions/ProjectionMappers/EnumerableProjectionMapper.cs

```diff
@@ -35,7 +36,19 @@ public Expression Project(IGlobalConfiguration configuration, MemberMap memberMa
             if (!memberMap.DestinationType.IsAssignableFrom(sourceExpression.Type))
             {
                 var convertFunction = memberMap.DestinationType.IsArray ? ToArrayMethod : ToListMethod;
-                sourceExpression = Call(convertFunction.MakeGenericMethod(destinationListType), sourceExpression);
+                convertFunction = convertFunction.MakeGenericMethod(destinationListType);
+                if (memberMap.DestinationType.IsAssignableFrom(convertFunction.ReturnType))
+                {
+                    sourceExpression = Call(convertFunction, sourceExpression);
+                }
+                else
+                {
+                    var ctorInfo = memberMap.DestinationType.GetConstructor(new[] { sourceExpression.Type });
+                    if (ctorInfo is not null)
+                    {
+                        sourceExpression = New(ctorInfo, sourceExpression);
```

Comment:
This probably needs a cast because the returned constructor parameter might not match exactly, but I'm not sure how likely is that to happen, so perhaps it's better to stick with the simpler code until proven otherwise.

Reply:
I think the cast will mostly fail, since `sourceExpression.Type` will be `IEnumerable<>` in the majority of the cases (product of Select statement). If we want to be more "loose" and support more specific ctor parameters, then perhaps the following will cover more cases?
```c#
var convertFunction = memberMap.DestinationType.IsArray ? ToArrayMethod : ToListMethod;
sourceExpression = Call(convertFunction.MakeGenericMethod(destinationListType), sourceExpression);
if (!memberMap.DestinationType.IsAssignableFrom(convertFunction.ReturnType))
{
    var ctorInfo = memberMap.DestinationType.GetConstructor(new[] { sourceExpression.Type });
    if (ctorInfo is not null)
    {
        sourceExpression = New(ctorInfo, sourceExpression);
    }
}
```

Reply:
I don't follow :) All I'm saying is that the constructor parameter can be a base type of `sourceExpression.Type` and then the `New` expression constructor will fail, but I'd rather see it fail before adding that ugly piece of code.

Reply:
Being the base type of it would be really rare. I meant the cases where ctor param is a derived type of `sourceExpression.Type`. The above code will have a better chance of success for those cases. The generated statement will be something like `new(__.ToList())`. Anyhow, not important :) Thanks for your help.
```c#
public class CustomList<T> : List<T>
{
    public CustomList(IList<T> collection) : base(collection)
    {
    }
}
```